docs/developer/decision-records/2022-06-28-entities-timestamp/README.md

```diff
@@ -0,0 +1,169 @@
+# Entities timestamp
+
+## Decision
+
+An entity represents any singular, identifiable and separate object inside an application.
+
+The decision based on the definition explained above is to add a timestamp to the EDC entities in their state of
+creation. These Entities are `ContractAgreement`, `ContractDefinition`, `PolicyDefinition`, and `Asset`.
+
+The entity classes `ContractNegotiation` and `TransferProcess` will extend the `StatefulEntity` class. The other
+entities will extend the `Entity` class. the `Entity` class will take the attributes `id` and `createdTimestamp` from
+`statefulEntity` and it will become its superclass.
+
+## Rationale
+
+As it was already done with the entities which extended the `StatefulEntity` class, where the entities can store the
+timestamp of their creation and also the last change of state timestamp.
+
+The idea of this change is to provide timestamps for creation of entities. This approach is similar to the entities that
+are extended from the class `StatefulEntity`
+
+## Approach
+
+Since some entities once are created, do not change their state, it is possible to differentiate them from the ones that
+do it. This is the reason why an abstraction called `Entity` was extracted from the class `StatefulEntity`.
+
+The class Entity will contain 2 attributes
+
+```java
+
+public abstract class Entity<T extends Entity<T>> {
+
+    protected String id;
+    protected long createdTimestamp;
+
+    protected Entity() {
+    }
+
+    ...
+}    
+```
+
+The class StatefulEntity will extend Entity
+
+```java
+public abstract class StatefulEntity<T extends StatefulEntity<T>> extends Entity implements TraceCarrier {
+
+    ...
+}
+
+```
+
+The timestamp will be measured as epoch seconds, using systemDefault (UTC) as ZoneId.
+
+```java
+
+Clock clock=Clock.systemDefaultZone();
```

Comment:
The `Clock` service is provided by the core of the EDC so it should be the only instance around, it's better to show that it can be injected and then used like
```
@Inject
private Clock clock;
```

```
var timestamp = clock.millis();
```